src/gui/qgsmaptoolidentify.cpp

```diff
@@ -206,54 +207,54 @@ void QgsMapToolIdentify::deactivate()
+    return identifyRasterLayer( results, qobject_cast<QgsRasterLayer *>( layer ), geometry, viewExtent, mapUnitsPerPixel, identifyContext );
   }
   else if ( layer->type() == QgsMapLayerType::VectorLayer && layerType.testFlag( VectorLayer ) )
   {
-    return identifyVectorLayer( results, qobject_cast<QgsVectorLayer *>( layer ), geometry );
+    return identifyVectorLayer( results, qobject_cast<QgsVectorLayer *>( layer ), geometry, identifyContext );
   }
   else if ( layer->type() == QgsMapLayerType::MeshLayer && layerType.testFlag( MeshLayer ) )
   {
-    return identifyMeshLayer( results, qobject_cast<QgsMeshLayer *>( layer ), geometry );
+    return identifyMeshLayer( results, qobject_cast<QgsMeshLayer *>( layer ), geometry, identifyContext );
   }
   else if ( layer->type() == QgsMapLayerType::VectorTileLayer && layerType.testFlag( VectorTileLayer ) )
   {
-    return identifyVectorTileLayer( results, qobject_cast<QgsVectorTileLayer *>( layer ), geometry );
+    return identifyVectorTileLayer( results, qobject_cast<QgsVectorTileLayer *>( layer ), geometry, identifyContext );
   }
   else
   {
     return false;
   }
 }
 
-bool QgsMapToolIdentify::identifyVectorLayer( QList<QgsMapToolIdentify::IdentifyResult> *results, QgsVectorLayer *layer, const QgsPointXY &point )
+bool QgsMapToolIdentify::identifyVectorLayer( QList<QgsMapToolIdentify::IdentifyResult> *results, QgsVectorLayer *layer, const QgsPointXY &point, const QgsIdentifyContext &identifyContext )
 {
-  return identifyVectorLayer( results, layer, QgsGeometry::fromPointXY( point ) );
+  return identifyVectorLayer( results, layer, QgsGeometry::fromPointXY( point ), identifyContext );
 }
 
-bool QgsMapToolIdentify::identifyMeshLayer( QList<QgsMapToolIdentify::IdentifyResult> *results, QgsMeshLayer *layer, const QgsGeometry &geometry )
+bool QgsMapToolIdentify::identifyMeshLayer( QList<QgsMapToolIdentify::IdentifyResult> *results, QgsMeshLayer *layer, const QgsGeometry &geometry, const QgsIdentifyContext &identifyContext )
 {
   const QgsPointXY point = geometry.asPoint();  // mesh layers currently only support identification by point
-  return identifyMeshLayer( results, layer, point );
+  return identifyMeshLayer( results, layer, point, identifyContext );
 }
 
-bool QgsMapToolIdentify::identifyMeshLayer( QList<QgsMapToolIdentify::IdentifyResult> *results, QgsMeshLayer *layer, const QgsPointXY &point )
+bool QgsMapToolIdentify::identifyMeshLayer( QList<QgsMapToolIdentify::IdentifyResult> *results, QgsMeshLayer *layer, const QgsPointXY &point, const QgsIdentifyContext &identifyContext )
 {
   QgsDebugMsgLevel( "point = " + point.toString(), 4 );
   if ( !layer )
     return false;
 
   double searchRadius = mOverrideCanvasSearchRadius < 0 ? searchRadiusMU( mCanvas ) : mOverrideCanvasSearchRadius;
-  bool isTemporal = mCanvas->mapSettings().isTemporal() && layer->temporalProperties()->isActive();
+  bool isTemporal = identifyContext.isTemporal() && layer->temporalProperties()->isActive();
```

Comment:
Also update line 264 to use the context's temporal range

Reply:
Oddly enough, this is in my local source tree, yet not in the initial commit. Dark voodoo at play here...